libminifi/include/utils/ResourceQueue.h

```diff
@@ -108,16 +108,16 @@ class ResourceQueue : public std::enable_shared_from_this<ResourceQueue<Resource
 
  private:
   void returnResource(std::unique_ptr<ResourceType> resource) {
-    logDebug("Returning [%p] resource", resource.get());
+    logDebug("Returning [{}] resource", static_cast<void*>(resource.get()));
     if (reset_fn_)
       reset_fn_.value()(*resource);
     internal_queue_.enqueue(std::move(resource));
   }
 
   template<typename ...Args>
-  void logDebug(const char * const format, Args&& ...args) {
+  void logDebug(fmt::format_string<std::invoke_result_t<decltype(core::logging::map_args), Args>...> fmt, Args&& ...args) {
```

Comment:
If this is going to be part of the public interface of `Logger`, then I'd add a public type alias for this metaprogramming magic, so users don't have to decipher all of this.

Reply:
Thats a good idea, changed it in https://github.com/apache/nifi-minifi-cpp/pull/1670/commits/0b82cf18a151400080017725bb97b925fe8d459d

Reply:
This particular instance was not changed to use the new type alias

Reply:
my bad, fixed in https://github.com/apache/nifi-minifi-cpp/pull/1670/commits/c0d52f87357aae54244b79a150820bef2dc06a52